sw/device/lib/dif/dif_keymgr.h

```diff
@@ -0,0 +1,683 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef OPENTITAN_SW_DEVICE_LIB_DIF_DIF_KEYMGR_H_
+#define OPENTITAN_SW_DEVICE_LIB_DIF_DIF_KEYMGR_H_
+
+/**
+ * @file
+ * @brief <a href="/hw/ip/keymgr/doc/">Key Manager</a> Device Interface
+ * Functions
+ */
+
+#include <stdint.h>
+
+#include "sw/device/lib/base/mmio.h"
+#include "sw/device/lib/dif/dif_warn_unused_result.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif  // __cplusplus
+
+/**
+ * Enumeration for enabling/disabling various functionality.
+ */
+typedef enum dif_keymgr_toggle {
+  /**
+   * Enabled state.
+   */
+  kDifKeymgrToggleEnable,
+  /**
+   * Disabled state.
+   */
+  kDifKeymgrToggleDisable,
+} dif_keymgr_toggle_t;
+
+/**
+ * Hardware instantiation parameters for key manager.
+ *
+ * This struct describes information about the underlying hardware that is
+ * not determined until the hardware design is used as part of a top-level
+ * design.
+ */
+typedef struct dif_keymgr_params {
+  /**
+   * Base address of key manager registers.
+   */
+  mmio_region_t base_addr;
+} dif_keymgr_params_t;
+
+/**
+ * A handle to key manager.
+ *
+ * This type should be treated as opaque by users.
+ */
+typedef struct dif_keymgr {
+  /**
+   * Hardware instantiation parameters.
+   */
+  dif_keymgr_params_t params;
+} dif_keymgr_t;
+
+/**
+ * Result of a key manager operation.
+ */
+typedef enum dif_keymgr_result {
+  /**
+   * The call succeeded.
+   */
+  kDifKeymgrOk = 0,
+  /**
+   * A non-specific error occurred and the hardware is in an invalid or
+   * irrecoverable state.
+   */
+  kDifKeymgrError = 1,
+  /**
+   * The caller supplied invalid arguments but the call did not cause any
+   * side-effects and the hardware is in a valid and recoverable state.
+   */
+  kDifKeymgrBadArg = 2,
+} dif_keymgr_result_t;
+
+/**
+ * Result of a key manager operation that writes to lockable registers.
+ */
+typedef enum dif_keymgr_lockable_result {
+  /**
+   * The call succeeded.
+   */
+  kDifKeymgrLockableOk = kDifKeymgrOk,
+  /**
+   * A non-specific error occurred and the hardware is in an invalid or
+   * irrecoverable state.
+   */
+  kDifKeymgrLockableError = kDifKeymgrError,
+  /**
+   * The caller supplied invalid arguments but the call did not cause any
+   * side-effects and the hardware is in a valid and recoverable state.
+   */
+  kDifKeymgrLockableBadArg = kDifKeymgrBadArg,
+  /**
+   * The register that needs to be written to is locked.
+   */
+  kDifKeyMgrLockableLocked,
+} dif_keymgr_lockable_result_t;
```

Comment:
I'm aware that this is currently missing a `Busy` constant. I need to confirm the behavior w/ the HW team.

Reply:
I don't think we need a separate enum for busy.